utils/node-loader/src/batch_pool/batch/upload_program.rs

```diff
@@ -27,10 +31,14 @@ impl UploadProgramArgs {
         let mut payload = vec![0; rng.gen_range(1..=100)];
         rng.fill_bytes(&mut payload);
 
-        // TODO: add this.
+        tracing::debug!(
+            "Generated `upload_program` batch with code seed = {code_seed}, salt = {} payload = {}",
+            hex::encode(&salt),
+            hex::encode(&payload)
+        );
+
         let gas_limit = 240_000_000_000;
```

Comment:
for all gas limits also caught it from the runtime please (see gclient/tests how to do it)